From 1e80734a44e5fb08fd31e068a8862aa27a442875 Mon Sep 17 00:00:00 2001
From: John Clark <inindev@gmail.com>
Date: Tue, 17 Oct 2023 02:29:18 +0000
Subject: [PATCH 2/2] board: starfive: edit scan_dev_for_boot_part to scan all
 partitions

Signed-off-by: John Clark <inindev@gmail.com>
---
 include/configs/starfive-visionfive2.h | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/include/configs/starfive-visionfive2.h b/include/configs/starfive-visionfive2.h
index ff43113f24..35db525440 100644
--- a/include/configs/starfive-visionfive2.h
+++ b/include/configs/starfive-visionfive2.h
@@ -46,8 +46,17 @@
 	"type_guid_gpt_loader1=" TYPE_GUID_SPL "\0" \
 	"type_guid_gpt_loader2=" TYPE_GUID_UBOOT "\0" \
 	"type_guid_gpt_system=" TYPE_GUID_SYSTEM "\0" \
-	"partitions=" PARTS_DEFAULT "\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
-	BOOTENV
+	BOOTENV \
+	"scan_dev_for_boot_part="                               \
+		"part list ${devtype} ${devnum} devplist; "     \
+			"env exists devplist || setenv devplist 1; "    \
+			"for distro_bootpart in ${devplist}; do "       \
+			"if fstype ${devtype} "                         \
+				"${devnum}:${distro_bootpart} "         \
+				"bootfstype; then "                     \
+				"run scan_dev_for_boot; "               \
+			"fi; "                                          \
+		"done\0"
 
 #endif /* _STARFIVE_VISIONFIVE2_H */
-- 
2.40.1

